### PR TITLE
split code by routes

### LINKED ISSUE #853: Implement React Router Code Splitting
Deliver only pieces of the application the user is using.

https://reactjs.org/docs/code-splitting.html#route-based-code-splitting

**Acceptance Criteria**
- [ ] Only download the minimum required front end javascript for each of our applications.

**Additional context**
-  We might get this for free if we switch vite or parcel bundler.

### PR DESCRIPTION
**Pull request by jmetev1**
1. code splitting on routes. closes #853 
the prod landing page gets a 78 score from chrome lighthouse on performance while the new dev landing page https://wps-pr-2944.apps.silver.devops.gov.bc.ca/ gets a 97. i'm not sure that reflects exactly what it will get on prod but i think so.
3. remove some unused files
previous bundle sizes: from https://github.com/bcgov/wps/actions/runs/5283902335/jobs/9560746213
File sizes after gzip:

  1.82 MB  build/static/js/main.2ef6db31.js
  7.85 kB  build/static/css/main.9ece5e76.css

new bundle sizes: 
File sizes after gzip:
  1.09 MB    build/static/js/557.aada5e04.chunk.js
  203.36 kB  build/static/js/main.61f6f2fb.js
  125.7 kB   build/static/js/916.317fd418.chunk.js
  89.53 kB   build/static/js/931.0e1edf45.chunk.js
  83.25 kB   build/static/js/834.abb41723.chunk.js
  66.97 kB   build/static/js/982.0761e93c.chunk.js
  35.88 kB   build/static/js/875.2e2adb97.chunk.js
  34.32 kB   build/static/js/338.a78ba776.chunk.js
  20.27 kB   build/static/js/568.732a9214.chunk.js
  16.23 kB   build/static/js/288.b24e6a14.chunk.js
  12.99 kB   build/static/js/23.ad75413a.chunk.js
  12.83 kB   build/static/js/381.73e2ecea.chunk.js
  10.69 kB   build/static/js/337.669d2df1.chunk.js
  10.1 kB    build/static/js/245.88dc54f9.chunk.js
  7.52 kB    build/static/js/740.867b4c1d.chunk.js
  6.55 kB    build/static/js/930.5dce4c79.chunk.js
  6.39 kB    build/static/css/465.525e2941.chunk.css
  5.94 kB    build/static/js/555.ec44b6ef.chunk.js
  5.8 kB     build/static/js/11.8b784ddd.chunk.js
  4.76 kB    build/static/js/470.ebc0126c.chunk.js
  4.88 kB    build/static/js/357.227a6445.chunk.js
  3.73 kB    build/static/js/689.441ba57c.chunk.js
  3.56 kB    build/static/js/535.fde67935.chunk.js
  3.46 kB    build/static/js/[194](https://github.com/bcgov/wps/actions/runs/5284604534/jobs/9562201645?pr=2944#step:4:194).5971816d.chunk.js
  3.14 kB    build/static/js/43.c04c7e05.chunk.js
  2.98 kB    build/static/js/465.383add64.chunk.js
  2.65 kB    build/static/js/226.7e7f5c18.chunk.js
  2.32 kB    build/static/js/216.833d3aa5.chunk.js
  1.7 kB     build/static/js/648.719c1fe0.chunk.js
  1.41 kB    build/static/css/288.1e011c12.chunk.css
  350 B      build/static/css/main.8914f316.css

The key here is that /main.61f6f2fb.js is much smaller so only people going to route with the big bundle will get it. i think. i would need to see the built files served to be sure which i can't do locally. maybe someone else can do when it's served from dev environment? i think main...js will be served to home page so that will load faster and then only one route will get the big bundle.

# Test Links:
[Landing Page](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-2944.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/app/Routes.tsx [hunk 1]
```diff
@@ -17,72 +19,74 @@ import {
...
-        <Route
-          path={HFI_CALC_ROUTE}
-          element={
-            <AuthWrapper>
-              <HfiCalculatorPage />
-            </AuthWrapper>
-          }
-        />
-        <Route path={C_HAINES_ROUTE} element={<CHainesPage />} />
+      <Suspense fallback={<div>Loading...</div>}>
```
Comment (jmetev1) [inline web/src/app/Routes.tsx:35]: this is just placeholder. do you want use like LoadingBackdrop.tsx or something else?

File: web/src/features/fba/components/map/FBAMap.tsx [hunk 2]
```diff
@@ -37,8 +37,6 @@ export const MapContext = React.createContext<ol.Map | null>(null)
 
 const zoom = 6
 const TILE_SERVER_URL = 'https://wps-prod-tileserv.apps.silver.devops.gov.bc.ca'
-export const SFMS_MAX_ZOOM = 8 // The SFMS data is so coarse, there's not much point in zooming in further
-export const COG_TILE_SIZE = [512, 512] // COG tiffs are 512x512 pixels - reading larger chunks should in theory be faster?
```
Comment (jmetev1) [inline web/src/features/fba/components/map/FBAMap.tsx:41]: had to move these around due to lazy loading

File: web/src/features/fba/pages/FireBehaviourAdvisoryPage.tsx [hunk 3]
```diff
@@ -64,7 +64,7 @@ const useStyles = makeStyles(() => ({
   }
 }))
 
-export const FireBehaviourAdvisoryPage: React.FunctionComponent = () => {
```
Comment (jmetev1) [inline web/src/features/fba/pages/FireBehaviourAdvisoryPage.tsx:67]: got rid of this bc lazy loading only works with default exports

File: web/src/app/Routes.tsx [hunk 4]
```diff
@@ -17,72 +19,75 @@ import {
...
-        <Route
-          path={HFI_CALC_ROUTE}
-          element={
-            <AuthWrapper>
-              <HfiCalculatorPage />
-            </AuthWrapper>
-          }
-        />
-        <Route path={C_HAINES_ROUTE} element={<CHainesPage />} />
+      <Suspense fallback={<LoadingBackdrop isLoadingWithoutError={true} />}>
```
Comment (jmetev1) [inline web/src/app/Routes.tsx:36]: i used the loadingbackdrop. is that good or you want something else?

Comment (conbrad) [reply web/src/app/Routes.tsx:36]: I think this is good thanks

**Review by conbrad — APPROVED** [review]
Nice work!

**Pull request merged**